spark/src/main/scala/io/substrait/spark/logical/ToSubstraitRel.scala

```diff
@@ -550,10 +550,12 @@ private[logical] class WithLogicalSubQuery(toSubstraitRel: ToSubstraitRel)
     expr match {
       case s: ScalarSubquery if s.outerAttrs.isEmpty && s.joinCond.isEmpty =>
         val rel = toSubstraitRel.visit(s.plan)
+        val t =
+          s.plan.schema.fields.head // Using this instead of s.dataType/s.nullable to get correct nullability
```

Comment:
There's two ScalarSubquery classes in Spark, the one we're using here is https://github.com/apache/spark/blob/9fe78e3d33499060467ecdc0c2631beae0b0316c/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/subquery.scala#L411 which always sets nullability to true. However the other one picks the nullability based on the plan's schema: https://github.com/apache/spark/blob/9fe78e3d33499060467ecdc0c2631beae0b0316c/sql/core/src/main/scala/org/apache/spark/sql/execution/subquery.scala#L68. I'm not very sure why there's a difference, but picking it from the plan seems to align better with what we expect here (and makes the tests pass)

Reply:
They both have `override def nullable: Boolean = true` (line 69 in your second link)?

Reply:
Woops indeed, I can't read 🤦. Still, I think the change here is fine nonetheless.